model/qti/metadata/ontology/OntologyMetadataInjector.php

```diff
@@ -26,53 +25,67 @@
 use oat\tao\model\event\MetadataModified;
 use oat\taoQtiItem\model\qti\metadata\MetadataInjector;
 use oat\taoQtiItem\model\qti\metadata\MetadataInjectionException;
-use \core_kernel_classes_Resource;
-use \core_kernel_classes_Property;
-use \InvalidArgumentException;
+use common_Logger;
+use core_kernel_classes_Property;
+use core_kernel_classes_Resource;
+use InvalidArgumentException;
+use Psr\Log\LoggerInterface;
 
 class OntologyMetadataInjector implements MetadataInjector
 {
     private $injectionRules;
-    
-    public function __construct()
-    {
+
+    /** @var EventManager */
+    private $eventManager;
+
+    /** @var LoggerInterface */
+    private $logger;
+
+    public function __construct(
+        LoggerInterface $logger = null,
+        EventManager $eventManager = null
+    ) {
+        $this->eventManager = $eventManager;
+        $this->logger = ($logger ?? common_Logger::singleton()->getLogger());
+
         $this->setInjectionRules([]);
     }
-    
+
     public function addInjectionRule(array $path, $propertyUri, $value = null, $ontologyValue = null)
     {
         if (count($path) === 0) {
-            $msg = "The path argument must be a non-empty array.";
-            throw new InvalidArgumentException($msg);
+            throw new InvalidArgumentException(
+                'The path argument must be a non-empty array.'
+            );
         }
-        
+
         $injectionRules = $this->getInjectionRules();
-        
+
         $pathKey = implode('->', $path);
         if (isset($injectionRules[$pathKey]) === false) {
             $injectionRules[$pathKey] = [];
         }
-        
+
         $injectionRules[$pathKey][] = [$propertyUri, $value, $ontologyValue];
         $this->setInjectionRules($injectionRules);
     }
-    
+
     protected function setInjectionRules(array $injectionRules)
     {
         $this->injectionRules = $injectionRules;
     }
-    
+
     protected function getInjectionRules()
     {
         return $this->injectionRules;
     }
-    
+
     public function inject($target, array $values)
     {
-        if (!$target instanceof core_kernel_classes_Resource) {
-            $msg = "The given target is not an instance of core_kernel_classes_Resource.";
-            throw new MetadataInjectionException($msg);
-        }
+        $this->assertIsResource($target);
+
+        $this->debug('values = %s', var_export($values, true));
```

Comment:
My idea was to keep them to ease debugging in the future in case it is needed. However, we may remove some of them (for example, this two in the top), or maybe keep just `'Remove property "%s" resource="%s" language="%s"'` & `'Set property "%s" value="%s" language="%s"'` ?

Reply:
Changes in this file have been reverted, so marking this thread as solved.